common/src/main/scala/net/psforever/objects/ExplosiveDeployable.scala

```diff
@@ -13,4 +26,86 @@ class ExplosiveDeployable(cdef : DeployableDefinition) extends SimpleDeployable(
     exploded = fuse
     Exploded
   }
+
+  override def Definition : ExplosiveDeployableDefinition = cdef
 }
+
+class ExplosiveDeployableDefinition(private val objectId : Int) extends ComplexDeployableDefinition(objectId) {
+  Name = "explosive_deployable"
+  DeployCategory = DeployableCategory.Mines
+  Model = StandardResolutions.SimpleDeployables
+  Packet = new SmallDeployableConverter
+
+  private var detonateOnJamming : Boolean = true
+
+  def DetonateOnJamming : Boolean = detonateOnJamming
+
+  def DetonateOnJamming_=(detonate : Boolean) : Boolean = {
+    detonateOnJamming = detonate
+    DetonateOnJamming
+  }
+
+  override def Initialize(obj : PlanetSideServerObject with Deployable, context : ActorContext) = {
+    obj.Actor = context.actorOf(Props(classOf[ExplosiveDeployableControl], obj), s"${obj.Definition.Name}_${obj.GUID.guid}")
+  }
+
+  override def Uninitialize(obj : PlanetSideServerObject with Deployable, context : ActorContext) = {
+    SimpleDeployableDefinition.SimpleUninitialize(obj, context)
+  }
+}
+
+object ExplosiveDeployableDefinition {
+  def apply(dtype : DeployedItem.Value) : ExplosiveDeployableDefinition = {
+    new ExplosiveDeployableDefinition(dtype.id)
+  }
+}
+
+class ExplosiveDeployableControl(mine : ExplosiveDeployable) extends Actor {
+  def receive : Receive = {
+    case Vitality.Damage(damage_func) =>
+      val originalHealth = mine.Health
+      if(originalHealth > 0) {
+        val cause = damage_func(mine)
+        ExplosiveDeployableControl.HandleDamageResolution(mine, cause, originalHealth - mine.Health)
+      }
+
+    case _ => ;
+  }
+}
+
+object ExplosiveDeployableControl {
+  def HandleDamageResolution(target : ExplosiveDeployable, cause : ResolvedProjectile, damage : Int) : Unit = {
+    val zone = target.Zone
+    val playerGUID = zone.LivePlayers.find { p => cause.projectile.owner.Name.equals(p.Name) } match {
+      case Some(player) => player.GUID
+      case _ => PlanetSideGUID(0)
+    }
+    if(target.Health == 0) {
+      HandleDestructionAwareness(target, playerGUID, cause)
+    }
+    else if(!target.Jammed && cause.projectile.profile.JammerProjectile) {
+      if(target.Jammed = {
```

Comment:
Isn't this always going to be false due to the check on the line above?

Reply:
`else if(!target.jammed && ...)` : this block only runs if we are not yet jammed.

`if(target.Jammed = { ... }` : set a value, the evaluation of the braced code, and return that value.  It may resolve to true and then we are jammed.

Reply:
Oops, I totally misread that as a comparison, not an assignment. As you were.